### DISCUSSION TITLE
Is there a way to return from infinit emscripten_set_main_loop

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#17155] [Q&A]
Hi,
Is there a way to return from infinit emscripten_set_main_loop? 

```
void loop(void) {
	emscripten_sleep(1);
	loops++;
	if (loops > 5) {
	emscripten_cancel_main_loop();
	}
}

void main() {
	emscripten_set_main_loop(loop, 1, 1);
        //we never get here
	int g=0;
        g++
}
```

**Comment by sbc100** [comment]
There is some documentation for emscripten_set_main_loop here: https://emscripten.org/docs/api_reference/emscripten.h.html#c.emscripten_set_main_loop

If you set simulate_infinite_loop to true then the code after the call to emscripten_set_main_loop is never run.  Its not possible return to that point in code.   From that point on the idea is that the loop runs instead of the code in main.

**Comment by aharondavid** [comment]
understood.
thanks!.
